cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java

```diff
@@ -500,6 +473,13 @@ public OutputStream getOutput() throws IOException {
   }
 
   /**
+   * Get the instance of the {@link HBaseDDLExecutor}.
+   * @param hConf the hadoop configurations
+   * @return the instance of DDL executor
+   */
+  public abstract HBaseDDLExecutor getHBaseDDLExecutor(Configuration hConf);
```

Comment:
One approach is we can inject the provider for the `HBaseDDLExecutor` in `HBaseTableUtil`. This provider will first check if there is any extension available(using `AbstractExtensionLoader`) if so will return that implementation, otherwise it will use `HBaseDDLExecutorFactory` to get the appropriate hbase version specific default executor.